moulinette/core.py

```diff
@@ -88,14 +88,23 @@ def translate(self, key, *args, **kwargs):
             - key -- The key to translate
 
         """
+        failed_to_format = False
         if key in self._translations.get(self.locale, {}):
-            return self._translations[self.locale][key].encode('utf-8').format(*args, **kwargs)
+            try:
+                return self._translations[self.locale][key].encode('utf-8').format(*args, **kwargs)
+            except KeyError as e:
+                logger.exception("Failed to format translated string '%s' with error: %s" % (key, e))
+                failed_to_format = True
 
-        if self.default_locale != self.locale and key in self._translations.get(self.default_locale, {}):
+        if failed_to_format or (self.default_locale != self.locale and key in self._translations.get(self.default_locale, {})):
             logger.info("untranslated key '%s' for locale '%s'",
```

Comment:
Hm that's no big deal, but this message is sometimes displayed even if the key is actually translated ...

for instance with locale as Fr : 

```
>>> m18n.n("user_unknown", app="test")
15044 WARNING Failed to format translated string 'user_unknown' with error: 'user'
15044 INFO untranslated key 'user_unknown' for locale 'fr'
15044 WARNING Failed to format translatable string 'user_unknown' with error: 'user'
'Utilisateur \xc2\xab {user:s} \xc2\xbb inconnu'
```

and a bit weirded with locale as En since it's the default locale : 

```
>>> m18n.n("user_unknown", app="test")
8407 WARNING Failed to format translated string 'user_unknown' with error: 'user'
8408 INFO untranslated key 'user_unknown' for locale 'en'
8408 WARNING Failed to format translatable string 'user_unknown' with error: 'user'
'Unknown user: {user:s}'
```